web-marketplace/src/pages/EcocreditsByAccount/CreditClassTab/CreditClassTab.tsx

```diff
@@ -1,49 +1,76 @@
-import { useMemo } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
 import { useLingui } from '@lingui/react';
-import { Grid } from '@mui/material';
 
-import { CreditClassGridCard } from 'web-components/src/components/molecules/CreditClassGridCard/CreditClassGridCard';
-import { LinkComponentType } from 'web-components/src/types/shared/linkComponentType';
+import { ImageActionCard } from 'web-components/src/components/cards/ImageActionCard';
+import { ProgramImageChildren } from 'web-components/src/components/cards/ProjectCard/ProjectCard.ImageChildren';
 
-import { getProjectCardBodyTextMapping } from 'lib/constants/shared.constants';
-import { useWallet } from 'lib/wallet/wallet';
-
-import { Link } from 'components/atoms';
 import WithLoader from 'components/atoms/WithLoader';
-import { useFetchCreditClassesWithOrder } from 'hooks/classes/useFetchCreditClassesWithOrder';
 
-import { useProfileData } from '../hooks/useProfileData';
+import { useMergedCreditClasses } from '../hooks/useMergedCreditClasses';
+import { LEARN_MORE_BUTTON } from './CreditClassTab.constants';
 
 export const CreditClassTab = () => {
-  const { _ } = useLingui();
-  const { wallet } = useWallet();
-  const { address } = useProfileData();
-  const { creditClasses, isLoadingCreditClasses } =
-    useFetchCreditClassesWithOrder({
-      admin: address,
-      userAddress: wallet?.address,
-    });
-  const bodyTexts = useMemo(() => getProjectCardBodyTextMapping(_), [_]);
+  const navigate = useNavigate();
+  const { accountAddressOrId } = useParams();
+  const { i18n } = useLingui();
+
+  const { creditClasses, programs, loading } =
+    useMergedCreditClasses(accountAddressOrId);
+
+  const extractTitle = (creditClass: any) => {
```

Comment:
ditto, let's avoid `any` and define a type for the credit classes returned by `useMergedCreditClasses`

we could also extract those functions into some CreditClassTab.utils.ts instead of having them in the component itself